Index: src/sys/kern/vfs_conf.c
===================================================================
RCS file: /cvs/src/sys/kern/vfs_conf.c,v
retrieving revision 1.33
diff -u -p -u -r1.33 vfs_conf.c
--- src/sys/kern/vfs_conf.c	2007/06/01 05:37:14	1.33
+++ src/sys/kern/vfs_conf.c	2007/11/07 22:55:54
@@ -119,6 +119,10 @@ extern  const struct vfsops ntfs_vfsops;
 extern  const struct vfsops udf_vfsops;
 #endif
 
+#ifdef O9FS
+extern	const struct vfsops o9fs_vfsops;
+#endif
+
 /*
  * Set up the filesystem operations for vnodes.
  */
@@ -177,6 +181,11 @@ static struct vfsconf vfsconflist[] = {
 	{ &udf_vfsops, MOUNT_UDF, 13, 0, MNT_LOCAL, NULL, NULL },
 #endif
 
+	/* O9FS Filesystem */
+#ifdef O9FS
+	{ &o9fs_vfsops, MOUNT_O9FS, 0, 0, 0, NULL, NULL },
+#endif
+
 };
 
 
@@ -216,6 +225,7 @@ extern struct vnodeopv_desc ext2fs_fifoo
 extern struct vnodeopv_desc xfs_vnodeop_opv_desc;
 extern struct vnodeopv_desc ntfs_vnodeop_opv_desc;
 extern struct vnodeopv_desc udf_vnodeop_opv_desc;
+extern struct vnodeopv_desc o9fs_vnodeop_opv_desc;
 
 struct vnodeopv_desc *vfs_opv_descs[] = {
 	&sync_vnodeop_opv_desc,
@@ -272,6 +282,9 @@ struct vnodeopv_desc *vfs_opv_descs[] = 
 #endif
 #ifdef UDF
 	&udf_vnodeop_opv_desc,
+#endif
+#ifdef O9FS
+	&o9fs_vnodeop_opv_desc,
 #endif
 
 	NULL
Index: src/sys/sys/mount.h
===================================================================
RCS file: /cvs/src/sys/sys/mount.h,v
retrieving revision 1.81
diff -u -p -u -r1.81 mount.h
--- src/sys/sys/mount.h	2007/06/01 05:37:14	1.81
+++ src/sys/sys/mount.h	2007/11/07 22:55:55
@@ -248,6 +248,13 @@ struct udf_args {
 	u_int32_t lastblock; /* Special device last block */
 };
 
+/* Argument to mount O9FS file systems */
+struct o9fs_args {
+	char	*hostname;	/* 9p server path */
+	struct sockaddr *saddr;	/* server addr */
+	size_t saddrlen;	/* addr size */	
+}; 
+	
 /*
  * Arguments to mount procfs filesystems
  */
@@ -342,6 +349,7 @@ struct ostatfs {
 #define	MOUNT_XFS	"xfs"		/* xfs */
 #define	MOUNT_NTFS	"ntfs"		/* NTFS */
 #define	MOUNT_UDF	"udf"		/* UDF */
+#define MOUNT_O9FS	"o9fs"		/* O9FS */
