Index: kern/vfs_conf.c
===================================================================
RCS file: /cvs/src/sys/kern/vfs_conf.c,v
retrieving revision 1.34
diff -u -p -r1.34 vfs_conf.c
--- kern/vfs_conf.c	28 Nov 2007 19:31:31 -0000	1.34
+++ kern/vfs_conf.c	30 Jan 2008 19:53:07 -0000
@@ -119,6 +119,10 @@ extern  const struct vfsops ntfs_vfsops;
 extern  const struct vfsops udf_vfsops;
 #endif
 
+#ifdef O9FS
+extern	const struct vfsops o9fs_vfsops;
+#endif
+
 /*
  * Set up the filesystem operations for vnodes.
  */
@@ -177,6 +181,11 @@ static struct vfsconf vfsconflist[] = {
 	{ &udf_vfsops, MOUNT_UDF, 13, 0, MNT_LOCAL, NULL, NULL },
 #endif
 
+	/* O9FS Filesystem */
+#ifdef O9FS
+	{ &o9fs_vfsops, MOUNT_O9FS, 0, 0, 0, NULL, NULL },
+#endif
+
 };
 
 
@@ -216,6 +225,7 @@ extern struct vnodeopv_desc ext2fs_fifoo
 extern struct vnodeopv_desc xfs_vnodeop_opv_desc;
 extern struct vnodeopv_desc ntfs_vnodeop_opv_desc;
 extern struct vnodeopv_desc udf_vnodeop_opv_desc;
+extern struct vnodeopv_desc o9fs_vnodeop_opv_desc;
 
 struct vnodeopv_desc *vfs_opv_descs[] = {
 	&sync_vnodeop_opv_desc,
@@ -272,6 +282,9 @@ struct vnodeopv_desc *vfs_opv_descs[] = 
 #endif
 #ifdef UDF
 	&udf_vnodeop_opv_desc,
+#endif
+#ifdef O9FS
+	&o9fs_vnodeop_opv_desc,
 #endif
 
 	NULL
Index: sys/malloc.h
===================================================================
RCS file: /cvs/src/sys/sys/malloc.h,v
retrieving revision 1.90
diff -u -p -r1.90 malloc.h
--- sys/malloc.h	28 Nov 2007 23:37:34 -0000	1.90
+++ sys/malloc.h	30 Jan 2008 19:53:08 -0000
@@ -183,7 +183,9 @@
 
 #define M_DRM		145	/* Direct Rendering Manager */
 
-#define	M_LAST		146	/* Must be last type + 1 */
+#define	M_O9FS		146	/* O9FS */
+
+#define	M_LAST		147	/* Must be last type + 1 */
 
 #define	INITKMEMNAMES { \
 	"free",		/* 0 M_FREE */ \
@@ -317,6 +319,7 @@
 	"Bluetooth HID",	/* 143 M_BTHIDEV */ \
 	"AGP Memory",	/* 144 M_AGP */ \
 	"Direct Rendering Manager",	/* 145 M_DRM */ \
+	"O9FS",		/* 146 M_O9FS */ \
 }
 
 struct kmemstats {
Index: sys/mount.h
===================================================================
RCS file: /cvs/src/sys/sys/mount.h,v
retrieving revision 1.81
diff -u -p -r1.81 mount.h
--- sys/mount.h	1 Jun 2007 05:37:14 -0000	1.81
+++ sys/mount.h	30 Jan 2008 19:53:09 -0000
@@ -248,6 +248,13 @@ struct udf_args {
 	u_int32_t lastblock; /* Special device last block */
 };
 
+/* Argument to mount O9FS file systems */
+struct o9fs_args {
+	char	*hostname;	/* 9p server path */
+	struct sockaddr *saddr;	/* server addr */
+	size_t saddrlen;	/* addr size */	
+}; 
+	
 /*
  * Arguments to mount procfs filesystems
  */
@@ -279,6 +286,7 @@ union mount_info {
 	struct procfs_args procfs_args;
 	struct msdosfs_args msdosfs_args;
 	struct ntfs_args ntfs_args;
+	struct o9fs_args o9fs_args;
 	char __align[160];	/* 64-bit alignment and room to grow */
 };
 
@@ -342,6 +350,7 @@ struct ostatfs {
 #define	MOUNT_XFS	"xfs"		/* xfs */
 #define	MOUNT_NTFS	"ntfs"		/* NTFS */
 #define	MOUNT_UDF	"udf"		/* UDF */
+#define MOUNT_O9FS	"o9fs"		/* O9FS */
 
 /*
  * Structure per mounted file system.  Each mounted file system has an
Index: sys/vnode.h
===================================================================
RCS file: /cvs/src/sys/sys/vnode.h,v
retrieving revision 1.90
diff -u -p -r1.90 vnode.h
--- sys/vnode.h	27 Dec 2007 13:59:12 -0000	1.90
+++ sys/vnode.h	30 Jan 2008 19:53:09 -0000
@@ -65,13 +65,13 @@ enum vtype	{ VNON, VREG, VDIR, VBLK, VCH
 enum vtagtype	{
 	VT_NON, VT_UFS, VT_NFS, VT_MFS, VT_MSDOSFS,
 	VT_PORTAL, VT_PROCFS, VT_AFS, VT_ISOFS, VT_ADOSFS,
-	VT_EXT2FS, VT_VFS, VT_XFS, VT_NTFS, VT_UDF
+	VT_EXT2FS, VT_VFS, VT_XFS, VT_NTFS, VT_UDF, VT_O9FS,
 };
 
 #define	VTAG_NAMES \
     "NON", "UFS", "NFS", "MFS", "MSDOSFS",			\
     "PORTAL", "PROCFS", "AFS", "ISOFS", "ADOSFS",		\
-    "EXT2FS", "VFS", "XFS", "NTFS", "UDF"
+    "EXT2FS", "VFS", "XFS", "NTFS", "UDF", "O9FS"
 
 /*
  * Each underlying filesystem allocates its own private area and hangs
